File: src/menu/app/src/main/java/net/rivergod/sec/seoulrnd/android/menu/dto/DayCuisionsDTO.java
package net.rivergod.sec.seoulrnd.android.menu.dto;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * Created by rivergod on 15. 12. 3.
 */
public class DayCuisionsDTO {

    private String date;
    private final Map<String, CuisineDTO> cuisines;

    public DayCuisionsDTO() {
        cuisines = new HashMap<>();
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public List<CuisineDTO> getCuisines() {
        return new ArrayList<CuisineDTO>(cuisines.values());
    }

    public boolean addCuisine(String id, CuisineDTO cuisine) {
        return cuisines.put(id, cuisine) != null;
    }

    public CuisineDTO getCuisineById(String id) {
        return cuisines.get(id);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof DayCuisionsDTO)) return false;

        DayCuisionsDTO that = (DayCuisionsDTO) o;

        if (date != null ? !date.equals(that.date) : that.date != null) return false;
        return cuisines.equals(that.cuisines);

    }

    @Override
    public int hashCode() {
        int result = date != null ? date.hashCode() : 0;
        result = 31 * result + cuisines.hashCode();
        return result;
    }

    @Override
    public String toString() {
        return "DayCuisionsDTO{" +
                "date='" + date + '\'' +
                ", cuisines=" + cuisines +
                '}';
    }
}

File: src/menu/app/src/main/kotlin/net/rivergod/sec/seoulrnd/android/menu/dto/DayCuisionsDTO.kt
package net.rivergod.sec.seoulrnd.android.menu.dto

data class DayCuisionsDTO(
    var date: String?,
    // Using LinkedHashMap to preserve insertion order like the original HashMap might,
    // though for DTO purposes, order from a HashMap isn't usually guaranteed or relied upon.
    // Switched to var to allow modification by addCuisine.
    var cuisines: MutableMap<String, CuisineDTO> = mutableMapOf()
) {

    // Constructor to match Java's default constructor behavior if needed,
    // though data classes get a primary constructor.
    // The default parameter for cuisines achieves a similar effect for default instantiation.
    // constructor() : this(null, mutableMapOf())

    fun getCuisines(): List<CuisineDTO> {
        return cuisines.values.toList()
    }

    fun addCuisine(id: String, cuisine: CuisineDTO): Boolean {
        return cuisines.put(id, cuisine) != null
    }

    fun getCuisineById(id: String): CuisineDTO? {
        return cuisines[id]
    }
}
